messageCheck/messageVerifier.go

```diff
@@ -140,7 +150,11 @@ func (m *messageVerifier) Serialize(messages []p2p.MessageP2P) ([]byte, error) {
 		pubsubMessages = append(pubsubMessages, pubsubMsgBytes)
 	}
 
-	messagesBytes, err := m.marshaller.Marshal(pubsubMessages)
+	if len(pubsubMessages) == 0 {
+		return []byte{}, nil
+	}
+
+	messagesBytes, err := json.Marshal(pubsubMessages)
```

Comment:
json is pretty slow.
Let's revert the usage of the internal marshal along with the [][]byte wrapper found in the elrond-go-core/data/batch

Reply:
changed, thanks for the suggestion